db/db.go

```diff
@@ -266,17 +262,17 @@ func (db *db) PrintDump(ctx context.Context) error {
 
 // Close is called when we are shutting down the database.
 // This is the place for any last minute cleanup or releasing of resources (i.e.: Badger instance).
-func (db *db) Close(ctx context.Context) {
-	log.Info(ctx, "Closing DefraDB process...")
+func (db *db) Close() {
+	log.Info(context.Background(), "Closing DefraDB process...")
```

Comment:
question: Was this what prompted the attempted removal of the `ctx` from the log interface?

thought: This does *feel* incorrect, as context has been defined outside of this function, and whilst at the moment as far as I can see they are all untouched `context.Background()`s, the context is not owned by this function, and could be modified by those owning layers, at which point this function would become not just conceptually incorrect, but tangibly so too.  It might be worth chatting about this - whilst a minor/unimportant part of the codebase, this kind of feels like a drop in standards and as such I was temped to mark this as a todo.

Reply:
Yeah this was the exact area where I ran into the log context. I plan on cleaning this up in the future when switching to the new `log/slog` go package.

Reply:
Okay, I'm happy enough if it is left as-is in the short term then if others are :)